### ISSUE TITLE
Is there a way to use query batching?

### ISSUE THREAD

**Issue by Sh1d0w** [zino-app/graphql-flutter#513]
Hi,

I see this plugin is similar to Apollo GraphQL for react which is great.

However I don’t see a BatchHttpLink available?

Is there a way to achieve query batching with this library?

**Comment by vladimirfx** [comment]
Any plans for this?
In medium+ sized projects query batching is must have. Multi component views generate its own queries witch transforms in multiple network requests per view. In mobile world with it latencies and connectivity quality have one request per view is vital usability factor. 
As workaround we can compromise application architecture and manually aggregate all needed data in one query/request. Such approach make Qraphql little difference with home grown REST API witch is having leaner learning curve. So usage of Graphql in Flutter is questionable as now.

Maybe there is another approach/workaround?

**Comment by canastro** [comment]
I reiterate @vladimirfx words, batching would be a key feature of graphql-flutter for the app I'm currently working.

I'm also interested in knowing any possible workarounds that wont compromise the app's architecture.

**Comment by vladimirfx** [comment]
@canastro exactly! Actually we discard GraphQL because lack of batching support on Flutter and switch to (partially) homemade WebSocket implementation with same semantic. One more critical thing for us is subscription support.

Anyway waiting for this feature for use in future projects.

**Comment by rwrz** [comment]
Well, no one seems to want this feature, is that right?
I believe it would be a GREAT addition. I'm adding here the ref. to the [Apollo's version](https://github.com/apollographql/apollo-client/blob/main/src/link/batch-http/batchHttpLink.ts).
I will add this to my plans, anyone here did something similar or started the work and have any insights of the problems we might face?
Thanks!
